src/main/java/school/hei/haapi/service/utils/ScholarshipCertificateDataProvider.java

```diff
@@ -4,28 +4,35 @@
 
 import java.time.ZoneId;
 import java.time.ZonedDateTime;
-import java.time.temporal.ChronoUnit;
 import lombok.AllArgsConstructor;
 import org.springframework.stereotype.Component;
 import school.hei.haapi.endpoint.rest.model.SpecializationField;
 import school.hei.haapi.model.User;
 import school.hei.haapi.model.exception.ApiException;
+import school.hei.haapi.service.PromotionService;
 
 @Component
 @AllArgsConstructor
 public class ScholarshipCertificateDataProvider {
   private final SchoolYearGetter schoolYearGetter;
+  private final IsStudentRepeatingYear isStudentRepeatingYear;
+  private final PromotionService promotionService;
 
   public String getAcademicYearSentence(User student) {
     String academicYear = getAcademicYear(student);
     return academicYear + " année d'informatique - parcours " + specializationFiledString(student);
   }
 
   public String getAcademicYear(User student) {
+    boolean isRepeating = isStudentRepeatingYear.apply(student);
     ZonedDateTime now = ZonedDateTime.now(ZoneId.of("UTC+3"));
     ZonedDateTime entranceDatetime = student.getEntranceDatetime().atZone(ZoneId.of("UTC+3"));
-    int year = (int) ChronoUnit.YEARS.between(entranceDatetime, now);
+    int year = now.getYear() - entranceDatetime.getYear();
 
+    if (isRepeating && year > 0) {
+      int numberOfRepeats = calculateNumberOfRepeats(student);
+      year -= numberOfRepeats;
+    }
```

Comment:
I think that the promotion startdatetime is more useable than the student entrance datetime, beacause the student is applicable only when he is in a group / in a promotion. If student is not in a group then this student is only register in HEI-admin but not study in HEI-admin